packages/react/src/components/Masthead/MastheadTopNav.js

```diff
@@ -23,6 +28,65 @@ const { prefix } = settings;
  * Masthead top nav component.
  */
 const MastheadTopNav = ({ navigation, ...topNavProps }) => {
+  const [showRightCaret, setShowRightCaret] = useState(false);
+  const [overflow, setOverflow] = useState(false);
+  const [showLeftCaret, setShowLeftCaret] = useState(false);
+
+  useEffect(() => {
+    onResize();
+    root.addEventListener('resize', onResize);
+    return () => {
+      root.removeEventListener('resize', onResize);
+    };
+  });
```

Comment:
Given we seem to want to track the size of `.bx--header__nav-container`, good to track its size directly via [`ResizeObserveer`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver).